### DISCUSSION TITLE
filed to connect to instance running on virtual vm

### DISCUSSION THREAD

**Discussion by vihariazure** [milvus-io/milvus#37913] [Q&A and General discussion]
Hi Team,

We have deployed Milvus on an Azure Virtual Machine using Docker. However, I am encountering an issue when trying to connect to the Milvus instance from my local machine.

Here is the Python code I am using:

python
Copy code
from pymilvus import connections

# Using host and port
connections.connect(
    host='xx.xx.xx.xx', 
    port='19530'
)
The following error occurs:

**javascript
Copy code
MilvusException: <MilvusException: (code=2, message=Fail connecting to server on xx.xx.xx.xx:19530, illegal connection params or server unavailable)>**

Steps I’ve Taken
I verified that Milvus is running in the VM and is listening on port 19530, as well as other services on ports 2379 and 9091.
I typically use PuTTY to log in to the server:
I specify the hostname or IP address in PuTTY, with the default port set to 22.
Once connected, I use a username and password, along with a ping ID for authentication.
We also use a VPN for accessing the server.
Despite these checks, the connection still fails. Any assistance or guidance on resolving this issue would be greatly appreciated.

**Comment by yhmo** [comment]
"illegal connection params or server unavailable" --- this error is thrown when grpc is trying to connect the remote address and timeout. By default the timeout setting is 10 seconds.
It is a network problem, it could not access xx.xx.xx.xx:19539 from the grpc layer. Maybe you can try turning off the firewall.
You can ssh to the host, and run the script to connect the milvus standalone from local. If connections.connect() works fine, then the milvus is healthy. If the connections.connect() also timeout on the host, you can collect the logs of the milvus by command "docker logs [container id] > milvus.log", there might be error logs in the file to reveal the root cause.

**Reply by vihariazure** [reply]
thanks for the reply
i tested on the vm. it is reachable. 
working with it team to open ports so that form out side also it is reachable. i will update you on this
